### PR TITLE
Fix H2H copy in HW NVJPEG.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
It fixes a host-to-host copy which specified a stream (now illegal) in hw-nvjpeg code path.

### Tests:
- [X] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [X] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h [hunk 1]
```diff
@@ -230,6 +230,10 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
     CUDA_CALL(cudaStreamCreateWithPriority(
       &hw_decode_stream_, cudaStreamNonBlocking, default_cuda_stream_priority_));
 
+    if (hw_decoder_images_staging_.is_pinned())
```
Comment (JanuszL) [inline dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:233]: I guess we can move it to the constructor but it should not make any difference after all.

Comment (JanuszL) [reply dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:233]: NVM

File: dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h [hunk 2]
```diff
@@ -269,10 +271,8 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
     });
     nvjpeg2k_thread_.RunAll();
 
-    CUDA_CALL(cudaStreamCreateWithPriority(&nvjpeg2k_cu_stream_, cudaStreamNonBlocking,
-                                           default_cuda_stream_priority_));
-    CUDA_CALL(cudaEventCreate(&nvjpeg2k_decode_event_));
-    CUDA_CALL(cudaEventRecord(nvjpeg2k_decode_event_, nvjpeg2k_cu_stream_));
+    nvjpeg2k_cu_stream_ = CUDAStreamPool::instance().Get();
+    nvjpeg2k_decode_event_ = CUDAEvent::Create();
```
Comment (JanuszL) [inline dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:275]: Shouldn't we recording the event as we used to?

**Review by JanuszL — APPROVED** [review]

File: dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h [hunk 3]
```diff
@@ -291,6 +291,10 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
 #if NVML_ENABLED
       nvml::Shutdown();
 #endif
+      if (hw_decode_stream_)
+        CUDA_CALL(cudaStreamSynchronize(hw_decode_stream_));
+
+      hw_decoder_images_staging_.Reset();
```
Comment (stiepan) [inline dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:297]: Why do we need it now?

File: dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h (hunk 2 above)
Comment (mzient) [reply dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h:275]: No; it was superfluous.

**Review by stiepan — APPROVED** [review]

**Pull request merged**